core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -121,19 +149,38 @@ impl BroadcastRun for StandardBroadcastRun {
         )
         .expect("Expected to create a new shredder");
 
-        let (data_shreds, coding_shreds, latest_shred_index) = shredder.entries_to_shreds(
+        let (data_shreds, coding_shreds, next_shred_index) = shredder.entries_to_shreds(
```

Comment:
I think there's a problem here. If last slot was interrupted, we compute `next_shred_index` here: https://github.com/solana-labs/solana/pull/6269/files#diff-4399f6663fb60984424c7221c5046bd6R105

which will be = to last index of interrupted slot. The last unfinished slot gets completed here: https://github.com/solana-labs/solana/pull/6269/files#diff-4399f6663fb60984424c7221c5046bd6R130

But, we are still using that index to shred the entries for the current slot. So entries for the current slot will start at an incorrect index.

Reply:
thanks, good catch!